wasm2c/examples/callback/main.c

```diff
@@ -26,7 +26,8 @@ int main(int argc, char** argv) {
    */
   wasm_rt_func_type_t fn_type =
       wasm2c_callback_get_func_type(1, 0, WASM_RT_I32);
-  wasm_rt_funcref_t fn_ref = {fn_type, (wasm_rt_function_ptr_t)print, &inst};
+  wasm_rt_funcref_t fn_ref = {
+      fn_type, (wasm_rt_function_ptr_t)print, {NULL}, &inst};
```

Comment:
It's a bit unfortunate that this leaks to the external types of the Wasm module versus being managed internally. This is probably fine, but would be nice to somehow hide from the host/embedder.

Reply:
Yeah... a challenge here is that our "module-to-host" interface is the same as the "module-to-module" interface. So if we want modules to be able to make tailcalls on imported functions from other modules, the tailcallee has to be part of the funcref type visible at this interface boundary. (And if we want the host to be able to make tailcalls to functions from a Wasm module, it also has to be part of the funcref type...)